js_modules/dagit/packages/core/src/runs/LogsProvider.tsx

```diff
@@ -237,11 +237,25 @@ const SubscriptionComponent = ({
   cursor: string | null;
   onSubscriptionData: (options: OnSubscriptionDataOptions<PipelineRunLogsSubscription>) => void;
 }) => {
+  const {availability, disabled, status} = React.useContext(WebSocketContext);
+  const lostWebsocket = !disabled && availability === 'available' && status === WebSocket.CLOSED;
+
+  const variables = React.useMemo(() => {
+    return {
+      runId,
+      cursor,
+    };
+    // Don't add cursor to memo otherwise subscription gets recreated
+    // Add lostWebSocket to dependency so that we record the cursor we're at
+    // for when the status comes back
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [runId, lostWebsocket]);
```

Comment:
Seems like you don't need to track `runId` here, since it shouldn't update.

Can this websocket observation behavior go in the `useLogsProviderWithSubscription`? Since that's where the logic lives to track the cursor value, it seems like that could also be the place where you could decide whether or not to set a new cursor to pass to the `SubscriptionComponent` and update its variables. Then maybe you don't need to hack around the hook rule.

What do you think?

Reply:
> Seems like you don't need to track runId here, since it shouldn't update.

From this component's point of view since its a prop it could potentially change though so I don't think it would be wise to use knowledge that it won't update that isn't codified into this component